src/io/EventReader.cc

```diff
@@ -22,8 +24,14 @@ EventReader::EventReader(const char* filename, SPConstParticles params)
     : params_(std::move(params))
 {
     CELER_EXPECT(params_);
+
+    // Turn off/capture HepMC3 diagnostic output that pollutes our own output
+    HepMC3::Setup::set_debug_level(-1);
+    ScopedStreamRedirect silenced(&std::cout);
+
     // Determine the input file format and construct the appropriate reader
     input_file_ = HepMC3::deduce_reader(filename);
+    CELER_LOG(diagnostic) << "HepMC3 said: " << silenced.str();
```

Comment:
If `set_debug_level` works as expected, `silenced` will be empty and this log message will be a bit silly.  Maybe
```suggestion
    std::string output = silenced.str();
    if (!output.empty())
    {
        CELER_LOG(diagnostic) << "HepMC3 said: " << output;
    }
```